master/internal/api_tensorboard.go

```diff
@@ -124,8 +117,20 @@ func (a *apiServer) GetTensorboard(
 func (a *apiServer) KillTensorboard(
 	ctx context.Context, req *apiv1.KillTensorboardRequest,
 ) (resp *apiv1.KillTensorboardResponse, err error) {
-	if _, err := a.GetTensorboard(ctx,
-		&apiv1.GetTensorboardRequest{TensorboardId: req.TensorboardId}); err != nil {
+	getResponse, err := a.GetTensorboard(ctx,
```

Comment:
I'm not sure I see what you mean - isn't the check that happens in `GetTensorboard` supposed to check for getting experiment artifacts?

Reply:
true I was thinking about it in terms of other NSC.
now that we do have separate RBAC interfaces for these different actions maybe the check for "can you terminate a tensorboard" would fail at `CanTerminateTensorboard` instead of `canGetTensorboard` and then the internal implementation could refer to `CanGetTensorboard` this would probably lead to better auditable logs 🤔